DataFormats/Detectors/TRD/src/RawData.cxx

```diff
@@ -192,60 +208,57 @@ uint32_t getChargeFromRawHeaders(const o2::trd::TrackletHCHeader& hcheader, cons
       //PID is 20 bits, 8 bits in mcmheader and 12 bits in mcmdata word
       //frist part of pid (highPID) is in the TrackletMCMHeader
       //highPID is 7 bits Q2, 1 bit Q1 OR ... 2 bit offset, 6 bits Q2.
-      switch (trackletindex) {
-        case 0:
-          highPID = header->pid0;
-          break;
-        case 1:
-          highPID = header->pid1;
-          break;
-        case 2:
-          highPID = header->pid2;
-          break;
-        default:
-          LOG(warn) << " unknown trackletindex of " << trackletindex << " to " << __func__ << " : " << pidindex;
-          break;
+      //trackletindex is the index into the [0:2] range of tracklets that can be attached to the mcm.
+      //jump to cpu:
+      uint32_t hpid = header->pid0 | (header->pid1 << 8) | (header->pid2 << 16);
+      std::array<uint16_t, 3> hpidvalues;
+      hpidvalues.fill(0xff);
+      uint16_t counter = 0;
+      if (header->pid0 != 0xff) {
+        hpidvalues[counter++] = header->pid0;
+      }
+      if (header->pid1 != 0xff) {
+        hpidvalues[counter++] = header->pid1;
+      }
+      if (header->pid2 != 0xff) {
+        hpidvalues[counter++] = header->pid2;
       }
+      //hpidvalues now holds the sequential list of high pid values related to the subsequent sequence of trackletmcmdata words
+      highPID = hpidvalues[trackletindex];
```

Comment:
This is not correct. In case CPU0 does not have a tracklet, but CPU1 does, you would still take the header PID information for the first tracklet from CPU0 (which in that case would be 0xff). I have a suggestion on how one could do it, we can check together later today

Reply:
but 0xff in pid first pid means the cpu0 does not have a tracklet. The first trackletword after the header is from cpu1.

Reply:
Indeed, since you fill the highPID only with the PID information when its not `0xff` this is also fine